Verification mods to feature/add metplus

michelleharrold opened the pull request:
## DESCRIPTION OF CHANGES: 
Modifications to:
scripts/exregional_get_mrms_files.sh --> changed logic to include tar file name change (dates prior to 20200304)
scripts/exregional_run_gridstatvx.sh --> added NET var for post file naming convention
scripts/exregional_run_pointstatvx.sh --> added NET var for post file naming convention
scripts/mrms_pull_topofhour.py --> changed logic to include tar file name change (dates prior to 20200304)
ush/templates/FV3.input.yml --> changed fhzero = 6.0 to fhzero = 1.0 for GFSv15.2 and CPT phys suites (necessary for hourly precip buckets)
ush/templates/parm/metplus/APCP_01h.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/APCP_03h.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/APCP_03h.conf_3hCCPA --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/APCP_06h.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/APCP_06h.conf_6hCCPA --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/APCP_24h.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/PointStat_conus_sfc.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/PointStat_upper_air.conf --> changed fcst file naming conventions for new post file name format
ush/templates/parm/metplus/REFC.conf --> changed fcst file naming conventions for new post file name format

## TESTS CONDUCTED: 
Ran modifications with GST for 2019061500 (48-h forecast) on Hera. End-to-end run was successful.

gsketefian committed:
Remove unused/outdated code (#313)

## DESCRIPTION OF CHANGES:
* In setup.sh and generate_FV3LAM_wflow.sh, remove temporary codes that fix bugs in the FV3_GFS_2017_gfdlmp_regional suite definition file because those bugs have been fixed (in the ufs-weather-model repo).
* In setup.sh, remove block of code that is no longer necessary because chgres_cube can now initialize from external model data with either 4 or 9 soil levels, and run with LSMs of either 4 or 9 soil levels.
* Remove modifications to LD_LIBRARY_PATH in exregional_run_fcst.sh.
* For the make_ics and make_lbcs tasks, move the setting of APRUN and other machine-specific actions from the J-job to the ex-script in order to be consistent with the other workflow tasks.
* Fix indentation and edit comments.
* Remove unused file load_fv3gfs_modules.sh.

## TESTS CONDUCTED: 
Ran two WE2E tests on hera, new_ESGgrid and new_GFDLgrid:
* new_ESGgrid uses the FV3_GFS_2017_gfdlmp_regional suite.  The test was successful.
* new_GFDLgrid uses the FV3_GFS_2017_gfdlmp suite.  The test was successful.

## ISSUE (optional): 
This resolves issue #198.

gsketefian committed:
Add and call a function that checks for use of Thompson microphysics parameterization in the SDF and if so, adjusts certain workflow arrays to contain the names and other associated values of the fixed files needed by this parameterization so that those files are automatically copied and/or linked to. (#319)

## DESCRIPTION OF CHANGES: 
Add and call a function that checks for use of Thompson microphysics parameterization in the suite definition file (SDF).  If not, do nothing.  If so, add to the appropriate workflow arrays the names and other associated values of the fixed files needed by this parameterization so that they are automatically copied and/or linked to instead of being regenerated from scratch in the run_fcst task.

## TESTS CONDUCTED: 
On hera, ran two WE2E tests, one in NCO mode (nco_RRFS_CONUS_25km_HRRRX_RAPX) and the other in community mode (suite_FV3_GSD_v0).  These use suites FV3_GSD_SAR and FV3_GSD_v0, respectively, and both of these call Thompson microphysics.  Both succeeded.

## ISSUE (optional):
This PR resolves issue #297.

mkavulich committed:
RRFS_v1beta SDF changes after reverting from GSL to GFS GWD suite (#322) (#327)

## DESCRIPTION OF CHANGES:
Removed checks on the RRFS_v1beta SDF implemented for use with the GSL GWD suite (now uses the GFS GWD suite).  No longer copies staged orography files necessary for the GSL GWD suite.

## TESTS CONDUCTED:
Runs to completion on Hera. End-to-end runs DOT_OR_USCORE and suite_FV3_RRFS_v1beta succeeded on Cheyenne.

Co-authored-by: JeffBeck-NOAA <55201531+JeffBeck-NOAA@users.noreply.github.com>

michelleharrold committed:
Update FV3.input.nml for fhzero = 1.0

michelleharrold committed:
Update FV3.input.nml for fhzero = 1.0

michelleharrold committed:
Updated conf files for file name conventions.

michelleharrold committed:
Updated conf files for file name conventions.

michelleharrold committed:
Updated MET scripts and MRMS pull scripts.

michelleharrold committed:
Updated MET scripts and MRMS pull scripts.

gsketefian committed:
Adjust RRFS_CONUS_... grids (#294)

## DESCRIPTION OF CHANGES: 
* Adjust RRFS_CONUS_25km, RRFS_CONUS_13km, and RRFS_CONUS_3km grid parameters so that:
  * All grids, including their 4-cell-wide halos, lie completely within the HRRRX domain.
  * All grids have dimensions nx and ny that factor "nicely", i.e. they don't have factors greather than 7.
  * The write-component grids corresponding to these three native grids cover as much of the native grids as possible without going outside of the native grid boundaries.  The updated NCL scripts (see below) were used to generate the write-component grid parameters.
* For the RRFS_CONUS_13km grid, reduce the time step (DT_ATMOS) from 180sec to 45sec.  This is necessary to get a successful forecast with the GSD_SAR suite, and thus likely also the RRFS_v1beta suite.
* Modify WE2E testing system as follows:
  * Add new tests with the RRFS_CONUS_25km, RRFS_CONUS_13km, and RRFS_CONUS_3km grids that use the GFS_v15p2 and RRFS_v1beta suites (which are now the ones officially supported in the first release of the short-range weather app) instead of the GFS_v16beta and GSD_SAR suites, respectively.
  * For clarity, rename the test configuration files that use the GFS_v16beta and GSD_SAR suites so they include the suite name.
  * Update list of WE2E tests (baselines_list.txt).
* Update the NCL plotting scripts to be able to plot grids with the latest version of the workflow.

## TESTS CONDUCTED: 
On hera, ran tests with all three grids with the GFS_v15p2 and RRFS_v1beta suites (a total of 6 tests).  All were successful.

jwolff-ncar commented on scripts/exregional_get_mrms_files.sh:
```diff
@@ -1,7 +1,9 @@
 #!/bin/sh
 
 # This script pulls MRMS data from the NOAA HPSS
-#OBS_DIR=/scratch2/BMC/det/jwolff/HIWT/obs/mrms/proc
```
Obviously I am guilty of included a personal path in here originally (and probably in other locations) but should we simply delete these commented out personal paths?

michelleharrold replied:
Yes, sorry for not deleting!

jwolff-ncar commented on scripts/exregional_get_mrms_files.sh:
```diff
@@ -22,97 +24,106 @@ if [[ ! -d "$mrms_proc" ]]; then
 # Initialization
-start_init=${CDATE}${hh}
-#start_init=2020063000
+start_valid=${CDATE}${hh}
```
Good catch. I guess I have been running this wrong with start_init!

jwolff-ncar commented on scripts/exregional_get_mrms_files.sh:
```diff
@@ -22,97 +24,106 @@ if [[ ! -d "$mrms_proc" ]]; then
   # Create necessary raw and proc directories
   if [[ ! -d "$mrms_raw/${vyyyymmdd}" ]]; then
     mkdir -p $mrms_raw/${vyyyymmdd}
+  fi
```
I assume you tested with this new indentation block and it works with and without the existence of a file? I have a hard time making that judgement just reading the code!

michelleharrold replied:
Yes (well, I ran it on two different dates to test, and both worked successfully; hopefully, that is good enough testing!).

jwolff-ncar replied:
One last follow up just to be clear. Did you test it twice with one date where the file would have already existed the second time? I am guessing it is totally fine.

michelleharrold replied:
I am pretty sure I did but couldn't 100% remember (lots of testing!). I just reran it with a date where the files existed, and it worked as expected.

jwolff-ncar commented on scripts/exregional_get_mrms_files.sh:
```diff
@@ -22,97 +24,106 @@ if [[ ! -d "$mrms_proc" ]]; then
-while [[ ${cur_ut} -le ${end_init_ut} ]]; do
-  cur_init=`date -ud '1970-01-01 UTC '${cur_ut}' seconds' +%Y%m%d%H` # convert universal time to standard time
-  echo "cur_init=${cur_init}"
+while [[ ${cur_ut} -le ${end_valid_ut} ]]; do
```
This is all still good with the python script and doesn't pull too much each time?

michelleharrold replied:
This change doesn't affect the original behavior (i.e., it pulls the same data as the original script).

jwolff-ncar commented on ush/templates/FV3.input.yml:
```diff
@@ -241,7 +241,7 @@ FV3_GFS_v15p2:
     fhlwr: 3600.0
     fhswr: 3600.0
-    fhzero: 6.0
+    fhzero: 1.0
```
Should we just remove this as is done in the other sections?

michelleharrold replied:
Yes, I meant to do that after testing. Good catch.

jwolff-ncar commented on scripts/exregional_run_gridstatvx.sh:
```diff
@@ -116,6 +117,7 @@ export MET_CONFIG
 export OBS_DIR
 export VAR
 export MODEL
+export NET
```
I'm fine with NET - does it stand for something? (I should probably know this)

michelleharrold replied:
I actually don't know what it stands for, but I noticed that the post run script uses it to name the post output. It is set and described in config_defaults.sh. I figured it would be good to use the preexisting post settings instead of creating a new variable...but that can definitely be changed!

jwolff-ncar replied:
Ha! Ok. Totally fine with reusing the variable. Was just curious what it was short for!

jwolff-ncar reviewed (commented):
I added a few comments in the files. I am not actually sure how they show up for you to review. Let me know if you can't see them!

gsketefian committed:
Remove redundant model_configure.${CCPP_PHYS_SUITE} template files; use Jinja2 to create model_configure (#321)

## DESCRIPTION OF CHANGES:
* Remove model_configure template files whose names depend on the physics suite, i.e. files with names of the form model_configure.${CCPP_PHYS_SUITE}.  Only a single template file is needed because the contents of the model_configure file are not suite dependent.  This leaves just one template file (named model_configure).
* Change the function create_model_configure_file.sh and the template file model_configure so they use jinja2 instead of sed to replace placeholder values.
* Absorb the contents of the write-component template files wrtcmp_lambert_conformal, wrtcmp_regional_latlon, and wrtcmp_rotated_latlon into the new jinja2-compliant model_configure file.  We can do this because Jinja2 allows use of if-statements in the template file.
* In the new model_configure jinja2 template file, include comments to explain the various write-component parameters.

## TESTS CONDUCTED:
On Hera, ran the two WE2E tests new_ESGgrid and new_GFDLgrid.  The first uses a "lambert_conformal" type of write-component grid, and the second uses a "rotated_latlon" type of write-component grid.  (The write-component also allows "regional_latlon" type grids, which is just the usual earth-relative latlon coordinate system, but we do not have any cases that use that.)  Both tests succeeded.

## ISSUE (optional): 
This PR resolves issue #281.

JeffBeck-NOAA committed:
Add Thompson ice- and water-friendly aerosol climo file support (#332)

* Add if statement in set_thompson_mp_fix_files.sh to source Thompson climo file when using a combination of a Thompson-based SDF and non-RAP/HRRR external model data

* Modify if statement based on external models for Thompson climo file

gsketefian committed:
Remove workflow variable EMC_GRID_NAME (#333)

## DESCRIPTION OF CHANGES: 
* Remove the workflow variable EMC_GRID_NAME.  Henceforth, PREDEF_GRID_NAME is the only variable that can be used to set the name of the predefined grid to use.
* Make appropriate change of variable name (EMC_GRID_NAME --> PREDEF_GRID_NAME) in the WE2E test configuration files.
* Change anywhere the "conus" and "conus_c96" grids are specified to "EMC_CONUS_3km" and "EMC_CONUS_coarse", respectively.
* Rename WE2E test configuration files with names containing the strings "conus" and "conus_c96" by replacing these strings with "EMC_CONUS_3km" and "EMC_CONUS_coarse", respectively.
* Update the list of WE2E test names (tests/baselines_list.txt).
* Bug fixes not directly related to grids:
  * In config.nco.sh, remove settings of QUEUE_DEFAULT, QUEUE_HPSS, and QUEUE_FCST since these are now set automatically (due to another PR).
  * In the template file FV3LAM_wflow.xml, add the ensemble member name after RUN_FCST_TN in the dependency of the run_post metatask.

## TESTS CONDUCTED: 
Since this change only affects runs in NCO mode, the following NCO-mode WE2E tests were rerun on hera, all successfully:
```
nco_EMC_CONUS_3km                                       SUCCESS
nco_EMC_CONUS_coarse                                    SUCCESS
nco_EMC_CONUS_coarse__suite_FV3_GFS_2017_gfdlmp         SUCCESS
nco_RRFS_CONUS_25km_HRRRX_RAPX                          SUCCESS
nco_RRFS_CONUS_3km_FV3GFS_FV3GFS                        SUCCESS
nco_RRFS_CONUS_3km_HRRRX_RAPX                           SUCCESS
nco_ensemble                                            SUCCESS
```

michelleharrold commented:
> I added a few comments in the files. I am not actually sure how they show up for you to review. Let me know if you can't see them!

I am probably just a total novice, but I don't know how to see your comments!

gsketefian committed:
Port workflow to Orion (#309)

## DESCRIPTION OF CHANGES:
* Add stanzas for Orion where necessary.
* Add new module files for Orion.
* On Orion, both the slurm partition and the slurm QOS need to be specified in the rocoto XML in order to be able to have wall times longer than 30 mins (the partition needs to be specified because it is by default "debug", which has a limit of 30 mins).  Thus, introduce modifications to more easily specify slurm partitions:
    * Remove the workflow variables QUEUE_DEFAULT_TAG, QUEUE_HPSS_TAG, and QUEUE_FCST_TAG that are currently used to determine whether QUEUE_DEFAULT, QUEUE_HPSS, and QUEUE_FCST specify the names of queue/QOS's or slurm partitions.
    * Add the workflow variables PARTITION_DEFAULT_TAG, PARTITION_HPSS_TAG, and PARTITION_FCST_TAG.  These will be used to specify slurm partitions only, and the variables QUEUE_DEFAULT, QUEUE_HPSS, and QUEUE_FCST will be used to specify queues/QOS's only.

IMPORTANT NOTE:
On Orion, in order to load the regional_workflow environment needed for generating an experiment, the user must first issue the following commands:
```
module use -a /apps/contrib/miniconda3-noaa-gsl/modulefiles
module load miniconda3
conda activate regional_workflow
```

## TESTS CONDUCTED:
Ran 11 WE2E tests on Orion, Hera, and Cheyenne.

Results on Orion:
```
community_ensemble_2mems          SUCCESS
DOT_OR_USCORE                     SUCCESS
grid_GSD_HRRR_AK_50km             FAILURE - In the run_fcst task.
  * Error message:
  !!! (1) Error in subr radiation_aerosols: unrealistic surface pressure =
           1                     NaN
new_ESGgrid                       SUCCESS
new_GFDLgrid                      SUCCESS
regional_001                      SUCCESS
regional_002                      SUCCESS
suite_FV3_GFS_v15p2               SUCCESS
suite_FV3_GFS_v16beta             SUCCESS
suite_FV3_GSD_SAR                 SUCCESS
suite_FV3_GSD_v0                  SUCCESS
```
Results on Hera:
```
community_ensemble_2mems          SUCCESS
DOT_OR_USCORE                     SUCCESS
grid_GSD_HRRR_AK_50km             SUCCESS
new_ESGgrid                       SUCCESS
new_GFDLgrid                      SUCCESS
regional_001                      SUCCESS
regional_002                      SUCCESS
suite_FV3_GFS_v15p2               SUCCESS
suite_FV3_GFS_v16beta             SUCCESS
suite_FV3_GSD_SAR                 SUCCESS
suite_FV3_GSD_v0                  SUCCESS
```
Results on Cheyenne:
```
community_ensemble_2mems          SUCCESS
DOT_OR_USCORE                     SUCCESS
grid_GSD_HRRR_AK_50km             FAILURE - In run_fcst task.
  * Error message:
  !!! (1) Error in subr radiation_aerosols: unrealistic surface pressure =
           1                     NaN
new_ESGgrid                       SUCCESS
new_GFDLgrid                      SUCCESS
regional_001                      SUCCESS
regional_002                      SUCCESS
suite_FV3_GFS_v15p2               SUCCESS
suite_FV3_GFS_v16beta             SUCCESS
suite_FV3_GSD_SAR                 SUCCESS
suite_FV3_GSD_v0                  SUCCESS
```
All succeed except GSD_HRRR_AK_50km on Orion and Cheyenne.  It is not clear why grid_GSD_HRRR_AK_50km fails on Orion and Cheyenne but not Hera.  Seems to point to a bug in the forecast model.  These two failures are not so important since this grid will soon be deprecated.

Also tested successfully on Jet by @JeffBeck-NOAA and on Odin and Stampede by @ywangwof.

## ISSUE:
This resolves Issue #152.

## CONTRIBUTORS:
@JeffBeck-NOAA @ywangwof @christinaholtNOAA

jwolff-ncar commented:
It looks like you saw my comments because you responded. I will go take one
last look and approve.

On Fri, Oct 23, 2020 at 8:24 PM michelleharrold <notifications@github.com>
wrote:

> I added a few comments in the files. I am not actually sure how they show
> up for you to review. Let me know if you can't see them!
>
> I am probably just a total novice, but I don't know how to see your
> comments!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jwolff-ncar/regional_workflow/pull/1#issuecomment-715658676>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFN5ZDG76IY2K2CN7COXRQDSMI3GTANCNFSM4SZ5L44Q>
> .
>


-- 

Jamie K. Wolff
National Center for Atmospheric Research (NCAR)
Research Applications Laboratory (RAL)
Phone: 303.497.2812  Email: jwolff@ucar.edu

*My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.*

jwolff-ncar commented:
Did you want to commit the changes for removing the commented out path and the fhzero setting? Then I can merge this in.

michelleharrold commented:
Oh, I thought you added some comments in the actual scripts. Got it.

I will commit the changes for removing the commented out path and the fhzero setting.

michelleharrold committed:
Removed comments from exregional_get_mrms_files.sh and removed fhzero from FV3.input.yml

michelleharrold committed:
Removed comments from exregional_get_mrms_files.sh and removed fhzero from FV3.input.yml

jwolff-ncar commented:
Great, thanks!

On Mon, Oct 26, 2020 at 9:35 AM michelleharrold <notifications@github.com>
wrote:

> *@michelleharrold* commented on this pull request.
> ------------------------------
>
> In scripts/exregional_get_mrms_files.sh
> <https://github.com/jwolff-ncar/regional_workflow/pull/1#discussion_r512056195>
> :
>
> >
>    # Create necessary raw and proc directories
>    if [[ ! -d "$mrms_raw/${vyyyymmdd}" ]]; then
>      mkdir -p $mrms_raw/${vyyyymmdd}
> +  fi
>
> I am pretty sure I did but couldn't 100% remember (lots of testing!). I
> just reran it with a date where the files existed, and it worked as
> expected.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jwolff-ncar/regional_workflow/pull/1#discussion_r512056195>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFN5ZDH6MFKVMNAGWTWV7ALSMWJN3ANCNFSM4SZ5L44Q>
> .
>


-- 

Jamie K. Wolff
National Center for Atmospheric Research (NCAR)
Research Applications Laboratory (RAL)
Phone: 303.497.2812  Email: jwolff@ucar.edu

*My working day may not be your working day. Please do not feel obliged to
reply to this email outside of your normal working hours.*

jwolff-ncar committed:
Update with develop

michelleharrold committed:
Merge branch 'feature/add_metplus' of https://github.com/michelleharrold/regional_workflow into feature/add_metplus
